plotting/plot_handler.py

```diff
@@ -0,0 +1,107 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Implements the PlotHandler class which acts as the controller for the plot functionality.
+This class is (will be) responsible for:
+Searching for and retrieving an existing plotting file via the SFTP client
+Getting the associated plotting meta data file
+Instructing the Plotting factory to build an IFrame based on the above
+"""
+import logging
+from utils.clients.sftp_client import SFTPClient
+
+LOGGER = logging.getLogger('app')
+
+
+# pylint:disable=line-too-long, no-else-return
+class PlotHandler:
+    """
+    Utility that takes parameters for a run and (for now) checks if an associated image exists and retrieves it.
+    :param instrument_name: The name of the beamline/spectrometer/instrument.
+    :param rb_number: The ISIS RB number.
+    :param run_number: The run number on the given instrument for the given RB number.
+    :param plot_type:
+        The type of plot file / file extension to be searched for (e.g. "png"). Can be a list of extensions.
+        If None, a list of common graphics file extensions is searched for.
+    """
+
+    def __init__(self, instrument_name, rb_number, run_number, plot_type=None):
+        self.instrument_name = instrument_name
+        self.rb_number = rb_number
+        self.run_number = run_number
+        if plot_type is None:
+            # list of plot types the handler looks for if none is provided; it searches for the types in the order of
+            # the list and selects the first one that exists. In future: could add vector graphics (.pdf, .eps)
+            self.plot_type = ["png", "jpg", "bmp", "gif", "tiff"]
```

Comment:
Might it be more accurate to include dots ('.') at the beginning of each string? Just for the sake of a more refined regex

Reply:
The regex already adds the dot. I chose this usage as it felt more naturally to me that someone might use the PlotHandler class thinking "I want to look for a png file" rather than "I want to look for a .png file". Either way is fine with me; the change would be easy (_regexp_for_file_name would have to be changed too)

Reply:
No problem, go ahead with the way that seems best to you 👍